src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractWorkspacePullDiagnosticsHandler.cs

```diff
@@ -130,6 +132,24 @@ protected override async Task WaitForChangesAsync(RequestContext context, Cancel
         return result.ToImmutable();
     }
 
+    /// <summary>
+    /// There are three potential sources for reporting workspace diagnostics:
+    ///
+    ///  1. Full solution analysis: If the user has enabled Full solution analysis, we always run analysis on the latest
+    ///                             project snapshot and return up-to-date diagnostics computed from this analysis.
+    ///
+    ///  2. Code analysis service: Otherwise, if full solution analysis is disabled, and if we have diagnostics from an explicitly
+    ///                            triggered code analysis execution on either the current or a prior project snapshot, we return
+    ///                            diagnostics from this execution. These diagnostics may be stale with respect to the current
+    ///                            project snapshot, but they match user's intent of not enabling continuous background analysis
+    ///                            for always having up-to-date workspace diagnostics, but instead computing them explicitly on
+    ///                            specific project snapshots by manually running the "Run Code Analysis" command on a project or solution.
+    ///
+    ///  3. EnC analysis: Emit and debugger diagnostics associated with a closed document or not associated with any document.
```

Comment:
nit.  tehse won't actually render.  can you use `<list>`

Reply:
I think it's ok if it doesn't render. It's still readable. `<list>` adds too much clutter, imo.